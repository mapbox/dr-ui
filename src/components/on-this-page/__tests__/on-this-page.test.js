import renderer from 'react-test-renderer';
import { testCases } from './on-this-page-test-cases.js';

describe('OnThisPage', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
