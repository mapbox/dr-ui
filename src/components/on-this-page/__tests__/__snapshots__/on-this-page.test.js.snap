// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnThisPage Basic renders as expected 1`] = `
<div
  className="limiter"
>
  <div
    className="grid grid--gut60"
  >
    <div
      className="col col--5-mxl col--12 relative dr-ui--page-layout-aside"
    >
      <div
        className="sticky-mxl top"
      >
        <nav
          aria-labelledby="on-this-page"
          className="dr-ui--on-this-page "
        >
          <h2
            className="unprose txt-m txt-bold mb6"
            id="on-this-page"
          >
            On this page
          </h2>
          <ul
            className=""
          >
            <li
              className="mb3 mb6-mxl"
            >
              <a
                className="link inline-block link--gray"
                href="#heading-1"
                style={
                  Object {
                    "wordBreak": "break-word",
                  }
                }
              >
                Heading 1
              </a>
              <ul
                className="none my6"
              >
                <li
                  className="mb3"
                >
                  <a
                    className="link inline-block link--gray"
                    href="#heading-a"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    Heading a
                  </a>
                </li>
                <li
                  className="mb3"
                >
                  <a
                    className="link inline-block link--gray"
                    href="#heading-b"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    Heading b
                  </a>
                </li>
                <li
                  className="mb3"
                >
                  <a
                    className="link inline-block link--gray"
                    href="#heading-c"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    Heading c
                  </a>
                </li>
              </ul>
            </li>
            <li
              className="mb3 mb6-mxl"
            >
              <a
                className="link inline-block link--gray"
                href="#heading-2"
                style={
                  Object {
                    "wordBreak": "break-word",
                  }
                }
              >
                Heading 2
              </a>
            </li>
            <li
              className="mb3 mb6-mxl"
            >
              <a
                className="link inline-block link--gray"
                href="#heading-3"
                style={
                  Object {
                    "wordBreak": "break-word",
                  }
                }
              >
                Heading 3
              </a>
              <ul
                className="none my6"
              >
                <li
                  className="mb3"
                >
                  <a
                    className="link inline-block link--gray"
                    href="#heading-a-1"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    <div
                      className="color-darken75 mr6 w18 h18 bg-gray-faint round-full flex-parent-inline flex-parent--center-main flex-parent--center-cross relative ml-neg12"
                      style={
                        Object {
                          "top": 3,
                        }
                      }
                    >
                      <svg
                        className="events-none icon"
                        focusable="false"
                        role="presentation"
                        style={
                          Object {
                            "height": 16,
                            "width": 16,
                          }
                        }
                      >
                        <use
                          xlinkHref="#icon-paint"
                          xmlnsXlink="http://www.w3.org/1999/xlink"
                        />
                      </svg>
                    </div>
                    Heading a 1
                  </a>
                </li>
                <li
                  className="mb3"
                >
                  <a
                    className="link inline-block link--gray"
                    href="#heading-b-1"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    <div
                      className="color-darken75 mr6 w18 h18 bg-gray-faint round-full flex-parent-inline flex-parent--center-main flex-parent--center-cross relative ml-neg12"
                      style={
                        Object {
                          "top": 3,
                        }
                      }
                    >
                      <svg
                        className="events-none icon"
                        focusable="false"
                        role="presentation"
                        style={
                          Object {
                            "height": 16,
                            "width": 16,
                          }
                        }
                      >
                        <use
                          xlinkHref="#icon-paint"
                          xmlnsXlink="http://www.w3.org/1999/xlink"
                        />
                      </svg>
                    </div>
                    Heading b 1
                  </a>
                </li>
                <li
                  className="mb3"
                >
                  <a
                    className="link inline-block link--gray"
                    href="#heading-c-1"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    <div
                      className="color-darken75 mr6 w18 h18 bg-gray-faint round-full flex-parent-inline flex-parent--center-main flex-parent--center-cross relative ml-neg12"
                      style={
                        Object {
                          "top": 3,
                        }
                      }
                    >
                      <svg
                        className="events-none icon"
                        focusable="false"
                        role="presentation"
                        style={
                          Object {
                            "height": 16,
                            "width": 16,
                          }
                        }
                      >
                        <use
                          xlinkHref="#icon-paint"
                          xmlnsXlink="http://www.w3.org/1999/xlink"
                        />
                      </svg>
                    </div>
                    Heading c 1
                  </a>
                </li>
              </ul>
            </li>
            <li
              className="mb3 mb6-mxl"
            >
              <a
                className="link inline-block link--gray"
                href="#heading-4"
                style={
                  Object {
                    "wordBreak": "break-word",
                  }
                }
              >
                Heading 4
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
    <div
      className="col col--7-mxl col--12 prose pb600"
    >
      <h2
        className="anchor"
        id="heading-1"
      >
        Heading 1
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <h3
        className="anchor"
        id="heading-a"
      >
        Heading a
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <h3
        className="anchor"
        id="heading-b"
      >
        Heading b
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <h3
        className="anchor"
        id="heading-c"
      >
        Heading c
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <h2
        className="anchor"
        id="heading-2"
      >
        Heading 2
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <h2
        className="anchor"
        id="heading-3"
      >
        Heading 3
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <h3
        className="anchor"
        id="heading-a-1"
      >
        Heading a 1
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <h3
        className="anchor"
        id="heading-b-1"
      >
        Heading b 1
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <h3
        className="anchor"
        id="heading-c-1"
      >
        Heading c 1
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <h2
        className="anchor"
        id="heading-4"
      >
        Heading 4
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
    </div>
  </div>
</div>
`;
